avalon/tools/cbsceneinventory/app.py

```diff
@@ -224,6 +239,199 @@ def show_remove_warning_dialog(self, items):
         self.data_changed.emit()
 
 
+class SwitchAssetDialog(QtWidgets.QDialog):
+    """Widget to support asset switching"""
+
+    switched = QtCore.Signal()
+
+    def __init__(self, parent=None, items=None):
+        QtWidgets.QDialog.__init__(self, parent)
+
+        self._items = items
+
+        self._assets_box = None
+        self._subsets_box = None
+        self._representations_box = None
+
+        input_layout = QtWidgets.QHBoxLayout()
+        input_layout.setDirection(QtWidgets.QBoxLayout.RightToLeft)
+
+        accept_icon = qta.icon("fa.check-square", color="white")
+        accept_btn = QtWidgets.QPushButton()
+        accept_btn.setIcon(accept_icon)
+        accept_btn.setFixedWidth(24)
+        accept_btn.setFixedHeight(24)
+
+        input_layout.addWidget(accept_btn)
+
+        self._input_layout = input_layout
+        self._accept_btn = accept_btn
+
+        self.setLayout(input_layout)
+        self.setWindowTitle("Switch selected items ...")
+
+        self.connections()
+
+        self.refresh()
+
+    def connections(self):
+        self._accept_btn.clicked.connect(self._on_accept)
+
+    def refresh(self):
+
+        assets = self._get_assets()
+        self._assets_box = self._create_combo_box(assets)
+
+        subsets = self._get_subsets()
+        self._subsets_box = self._create_combo_box(subsets)
+
+        representations = self._get_representations()
+        self._representations_box = self._create_combo_box(representations)
+
+        self._input_layout.addWidget(self._representations_box)
```

Comment:
Oh, no. This seems invalid. Another call to refresh will add another combobox, hehe. You'll end up with a lot of comboboxes. ;)

The layout is static - the comboboxes should just be there, they shouldn't be generated each time. They just are repopulated with new data.

Reply:
layout is static, correct but I am lazy and I wanted to ensure that I only need to code the creation of the combo box ONCE! Another plus side is that, if we need a functionality in all comboboxes we have it covered with one function ;)

Reply:
Your laziness is breaking your logic. Just do this:

```python
        self._assets_box = QtWidgets.QComboBox()
        self._subsets_box = QtWidgets.QComboBox()
        self._representations_box = QtWidgets.QComboBox()
```` 

And change the other method to:
```python
    def _populate_combo_box(self, combobox, items):
        """Repopulate the combobox with auto completion, first item will be "----"

        Args:
            items (iterable): list of document names
        """
        completer = QtWidgets.QCompleter(items)
        combobox.setCompleter(completer)
        combobox.setEditable(True)
        combobox.clear() # or whatever the method is called.
        combobox.addItem("----")
        combobox.addItems(items)
```

Still lazy.